From ff67674f4df8779923da2ff9d8f94da0857dfb50 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Mon, 8 Oct 2018 11:11:41 +0200
Subject: [PATCH 4/5] pie-gsi-tracking

commit 5f5584f076a80743e2a3db83aec10da0b8ed58ea
Author: Tri Vo <trong@google.com>
Date:   Thu Sep 20 11:05:09 2018 -0700

    Make dalvik.vm.boot-dex2oat-threads vendor-init-settable.

    Bug: 115741899
    Test: m selinux_policy
    Change-Id: I5d80a1d9bd5500a82ebf282fb02f0db3a0b0a4c1
    Merged-In: I5d80a1d9bd5500a82ebf282fb02f0db3a0b0a4c1
    (cherry picked from commit 260a275836ec32dd6f4a9422045b6d9d957c6af4)

commit 974aaa578a5eac96fb1921a8968d345ced79bf0a
Author: Tri Vo <trong@google.com>
Date:   Mon Sep 17 19:17:41 2018 -0700

    system_writes_mnt_vendor_violators for device launched before P.

    In cases when a device upgrades to system-as-root from O to P, it needs a mount
    point for an already existing partition that is accessed by both system and
    vendor.

    Devices launching with P must not have /mnt/vendor accessible to system.

    Bug: 78598545
    Test: m selinx_policy
    Change-Id: Ia7bcde44e2b8657a7ad9e0d9bae7a7259f40936f
    Merged-In: Ia7bcde44e2b8657a7ad9e0d9bae7a7259f40936f
    (cherry picked from commit ca4217e2111cb8e5ca8cae7a6a923cd5d5408e33)

commit 8bcf94a97eb7366788eab019a3cc9ff3ed3ebcbd
Author: Bowgo Tsai <bowgotsai@google.com>
Date:   Fri Jul 13 23:34:48 2018 +0800

    Allowing vold to search /mnt/vendor/*

    vold will trim rw mount points about daily, but it is denied by SELinux:

    root   603   603 W Binder:603_2: type=1400 audit(0.0:11): avc: denied {
    search } for name="vendor" dev="tmpfs" ino=23935 scontext=u:r:vold:s0
    tcontext=u:object_r:mnt_vendor_file:s0 tclass=dir permissive=0

    Allowing vold to search /mnt/vendor/* to fix the denials.

    Note that device-specific sepolicy needs to be extended to allow vold
    to send FITRIM ioctl. e.g., for /mnt/vendor/persist, it needs:

        allow vold persist_file:dir { ioctl open read };

    Bug: 111409607
    Test: boot a device, checks the above denial is gone
    Change-Id: Ia9f22d973e5a2e295678781de49a0f61fccd9dad
    Merged-In: Ia9f22d973e5a2e295678781de49a0f61fccd9dad
    (cherry picked from commit 7b67a617dd498e8b892c3475e96deaa0da6d1649)

commit f10f011d965ca975357214d51e34ba3052c0f332
Author: padarshr <padarshr@codeaurora.org>
Date:   Wed Jun 13 17:20:34 2018 +0530

    Add ueventd to mnt_vendor_file neverallow exemption list

    Legacy hardware and code still depends on the ueventd helpers to
    locate the firmware supported files which are on new mount path
    labeled with mnt_vendot_file. For ueventd helper to work we need dir search
    and read permission on this new label so moving ueventd to exempted list.

    Already ueventd has the vendor_file_type read access.

    Bug:110083808
    Test: boot a device

    Change-Id: Ia15cc39ecef9e29b4f1f684efdddbeb78b427988
    Merged-In: Ia15cc39ecef9e29b4f1f684efdddbeb78b427988
    (cherry picked from commit 44ae7c2ccb5b2e31eeaa2ed091c4d9d543a8294c)

commit f25ed2f2e9977bd651f6f709140eb337f1a4e54c
Merge: d6ee5db5 25fe3474
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Sep 14 22:21:51 2018 -0700

    Merge cherrypicks of [4647037, 4647038, 4647883, 4647039, 4647933, 4648530, 4648550, 4648551, 4648552, 4648553, 4646931, 4646932, 4646933, 4646934, 4648391, 4647976, 4647977, 4647978, 4647526, 4646972, 4646935, 4646936, 4646937, 4646938, 4646939, 4646940, 4646941, 4648392, 4647509, 4648630, 4648631, 4647934] into pi-release-2
    am: 25fe3474a1

    Change-Id: Ieef1da84d907a308ffcc45e6713dd01e56a6f83c

commit d6ee5db590287095f2d44e49e7caf7c400de58a0
Author: Bowgo Tsai <bowgotsai@google.com>
Date:   Fri Sep 14 12:49:14 2018 +0800

    Fix build break

    Need to update prebuilts/api/28.0 to fix the build error caused by:

      https://android-review.googlesource.com/c/platform/system/sepolicy/+/741207

    Bug: 113238007
    Test: diff -rq system/sepolicy/prebuilts/api/28.0/private system/sepolicy/private
    Change-Id: I83c3607f6a530f2fb842aaf39c123897b243f78f

commit e76fd88294d7d8f388232dce8240606fcdaea9bc
Author: Bowgo Tsai <bowgotsai@google.com>
Date:   Fri Sep 14 10:48:06 2018 +0800

    Fix build break

    Need to update prebuilts/api/28.0 to fix the build error caused by:

      https://android-review.googlesource.com/c/platform/system/sepolicy/+/751394

    Bug: 114017832
    Test: m selinux_policy
    Change-Id: I077a368b33682d90bdb1a007ba071708ee70ae42

commit edc5776718cb294712acf1471c8afbe44e6b7d5f
Author: David Ng <dave@codeaurora.org>
Date:   Wed Apr 11 10:43:57 2018 -0700

    Explicitly allow system_server to (m)map data files

    Linux kernel 4.14+ SELinux starts explicit map
    permission check for file mmap operations.  Add this
    permission to system_server for data file access,
    which is used in scenario such as "adb install" of
    APK's.

    test: no longer see SELinux map denial on "adb install"
    Bug: 113238007
    Change-Id: Id6016dd0b3f15dfdb0f02509ea812dee61ac78ed
    Merged-In: Id6016dd0b3f15dfdb0f02509ea812dee61ac78ed
    (cherry picked from commit 383471c267b6792f1625f8f771d8e0c0b9090300)

commit b0226eee5d340e01240f13225a4ed890e2aa63cc
Author: Tri Vo <trong@google.com>
Date:   Sat Sep 8 14:42:49 2018 -0700

    ro.crypto.{allow_encrypt_override filenames_mode} vendor-init-settable.

    Bug: 114017832
    Test: m selinux_policy
    Change-Id: I1dcb09c76b3e49888d278a154d79add6c6a6c977
    Merged-In: I1dcb09c76b3e49888d278a154d79add6c6a6c977
    (cherry picked from commit fe72cb70d75479d726ae090ea871a7de943bad41)

Change-Id: Iff060645cc1a3f8f1a6fc5ab3ad3e639c5cc81a2
---
 prebuilts/api/28.0/private/system_server.te | 9 ++++++++-
 prebuilts/api/28.0/public/attributes        | 5 +++++
 prebuilts/api/28.0/public/domain.te         | 3 +++
 prebuilts/api/28.0/public/property_contexts | 3 +++
 prebuilts/api/28.0/public/vold.te           | 3 +++
 private/system_server.te                    | 9 ++++++++-
 public/attributes                           | 5 +++++
 public/domain.te                            | 3 +++
 public/property_contexts                    | 3 +++
 public/vold.te                              | 3 +++
 10 files changed, 44 insertions(+), 2 deletions(-)

diff --git a/prebuilts/api/28.0/private/system_server.te b/prebuilts/api/28.0/private/system_server.te
index fa84c322..ceae61b3 100644
--- a/prebuilts/api/28.0/private/system_server.te
+++ b/prebuilts/api/28.0/private/system_server.te
@@ -455,7 +455,14 @@ allow system_server system_app_data_file:file create_file_perms;
 
 # Receive and use open app data files passed over binder IPC.
 # Types extracted from seapp_contexts type= fields.
-allow system_server { system_app_data_file bluetooth_data_file nfc_data_file radio_data_file shell_data_file app_data_file }:file { getattr read write append };
+allow system_server {
+  system_app_data_file
+  bluetooth_data_file
+  nfc_data_file
+  radio_data_file
+  shell_data_file
+  app_data_file
+}:file { getattr read write append map };
 
 # Access to /data/media for measuring disk usage.
 allow system_server media_rw_data_file:dir { search getattr open read };
diff --git a/prebuilts/api/28.0/public/attributes b/prebuilts/api/28.0/public/attributes
index 0c7ca2ed..42b61067 100644
--- a/prebuilts/api/28.0/public/attributes
+++ b/prebuilts/api/28.0/public/attributes
@@ -177,6 +177,11 @@ expandattribute system_executes_vendor_violators false;
 attribute system_writes_vendor_properties_violators;
 expandattribute system_writes_vendor_properties_violators false;
 
+# All system domains which violate the requirement of not writing to
+# /mnt/vendor/*. Must not be used on devices launched with P or later.
+attribute system_writes_mnt_vendor_violators;
+expandattribute system_writes_mnt_vendor_violators false;
+
 # hwservices that are accessible from untrusted applications
 # WARNING: Use of this attribute should be avoided unless
 # absolutely necessary.  It is a temporary allowance to aid the
diff --git a/prebuilts/api/28.0/public/domain.te b/prebuilts/api/28.0/public/domain.te
index 1fe01d79..fe03c95d 100644
--- a/prebuilts/api/28.0/public/domain.te
+++ b/prebuilts/api/28.0/public/domain.te
@@ -1396,6 +1396,9 @@ userdebug_or_eng(`
 neverallow {
   coredomain
   -init
+  -ueventd
+  -vold
+  -system_writes_mnt_vendor_violators
 } mnt_vendor_file:dir *;
 
 # Vendor domian must not have access to /mnt/product.
diff --git a/prebuilts/api/28.0/public/property_contexts b/prebuilts/api/28.0/public/property_contexts
index 4f81c1c3..8e53eef8 100644
--- a/prebuilts/api/28.0/public/property_contexts
+++ b/prebuilts/api/28.0/public/property_contexts
@@ -8,6 +8,7 @@ camera.disable_zsl_mode u:object_r:exported3_default_prop:s0 exact bool
 camera.fifo.disable u:object_r:exported3_default_prop:s0 exact int
 dalvik.vm.appimageformat u:object_r:exported_dalvik_prop:s0 exact string
 dalvik.vm.backgroundgctype u:object_r:exported_dalvik_prop:s0 exact string
+dalvik.vm.boot-dex2oat-threads u:object_r:exported_dalvik_prop:s0 exact int
 dalvik.vm.checkjni u:object_r:exported_dalvik_prop:s0 exact bool
 dalvik.vm.dex2oat-Xms u:object_r:exported_dalvik_prop:s0 exact string
 dalvik.vm.dex2oat-Xmx u:object_r:exported_dalvik_prop:s0 exact string
@@ -97,7 +98,9 @@ ro.config.notification_sound u:object_r:exported2_config_prop:s0 exact string
 ro.config.ringtone u:object_r:exported2_config_prop:s0 exact string
 ro.control_privapp_permissions u:object_r:exported3_default_prop:s0 exact string
 ro.cp_system_other_odex u:object_r:exported3_default_prop:s0 exact int
+ro.crypto.allow_encrypt_override u:object_r:exported2_vold_prop:s0 exact bool
 ro.crypto.scrypt_params u:object_r:exported2_vold_prop:s0 exact string
+ro.crypto.volume.filenames_mode u:object_r:exported2_vold_prop:s0 exact string
 ro.dalvik.vm.native.bridge u:object_r:exported_dalvik_prop:s0 exact string
 ro.enable_boot_charger_mode u:object_r:exported3_default_prop:s0 exact bool
 ro.gfx.driver.0 u:object_r:exported3_default_prop:s0 exact string
diff --git a/prebuilts/api/28.0/public/vold.te b/prebuilts/api/28.0/public/vold.te
index 131f555d..4d15f11e 100644
--- a/prebuilts/api/28.0/public/vold.te
+++ b/prebuilts/api/28.0/public/vold.te
@@ -206,6 +206,9 @@ allow vold user_profile_data_file:dir create_dir_perms;
 # Raw writes to misc block device
 allow vold misc_block_device:blk_file w_file_perms;
 
+# vold might need to search or mount /mnt/vendor/*
+allow vold mnt_vendor_file:dir search;
+
 neverallow {
     domain
     -vold
diff --git a/private/system_server.te b/private/system_server.te
index fa84c322..ceae61b3 100644
--- a/private/system_server.te
+++ b/private/system_server.te
@@ -455,7 +455,14 @@ allow system_server system_app_data_file:file create_file_perms;
 
 # Receive and use open app data files passed over binder IPC.
 # Types extracted from seapp_contexts type= fields.
-allow system_server { system_app_data_file bluetooth_data_file nfc_data_file radio_data_file shell_data_file app_data_file }:file { getattr read write append };
+allow system_server {
+  system_app_data_file
+  bluetooth_data_file
+  nfc_data_file
+  radio_data_file
+  shell_data_file
+  app_data_file
+}:file { getattr read write append map };
 
 # Access to /data/media for measuring disk usage.
 allow system_server media_rw_data_file:dir { search getattr open read };
diff --git a/public/attributes b/public/attributes
index 0c7ca2ed..42b61067 100644
--- a/public/attributes
+++ b/public/attributes
@@ -177,6 +177,11 @@ expandattribute system_executes_vendor_violators false;
 attribute system_writes_vendor_properties_violators;
 expandattribute system_writes_vendor_properties_violators false;
 
+# All system domains which violate the requirement of not writing to
+# /mnt/vendor/*. Must not be used on devices launched with P or later.
+attribute system_writes_mnt_vendor_violators;
+expandattribute system_writes_mnt_vendor_violators false;
+
 # hwservices that are accessible from untrusted applications
 # WARNING: Use of this attribute should be avoided unless
 # absolutely necessary.  It is a temporary allowance to aid the
diff --git a/public/domain.te b/public/domain.te
index 374f050e..5131c2b0 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -1412,6 +1412,9 @@ userdebug_or_eng(`
 neverallow {
   coredomain
   -init
+  -ueventd
+  -vold
+  -system_writes_mnt_vendor_violators
 } mnt_vendor_file:dir *;
 
 # Vendor domian must not have access to /mnt/product.
diff --git a/public/property_contexts b/public/property_contexts
index 4f81c1c3..8e53eef8 100644
--- a/public/property_contexts
+++ b/public/property_contexts
@@ -8,6 +8,7 @@ camera.disable_zsl_mode u:object_r:exported3_default_prop:s0 exact bool
 camera.fifo.disable u:object_r:exported3_default_prop:s0 exact int
 dalvik.vm.appimageformat u:object_r:exported_dalvik_prop:s0 exact string
 dalvik.vm.backgroundgctype u:object_r:exported_dalvik_prop:s0 exact string
+dalvik.vm.boot-dex2oat-threads u:object_r:exported_dalvik_prop:s0 exact int
 dalvik.vm.checkjni u:object_r:exported_dalvik_prop:s0 exact bool
 dalvik.vm.dex2oat-Xms u:object_r:exported_dalvik_prop:s0 exact string
 dalvik.vm.dex2oat-Xmx u:object_r:exported_dalvik_prop:s0 exact string
@@ -97,7 +98,9 @@ ro.config.notification_sound u:object_r:exported2_config_prop:s0 exact string
 ro.config.ringtone u:object_r:exported2_config_prop:s0 exact string
 ro.control_privapp_permissions u:object_r:exported3_default_prop:s0 exact string
 ro.cp_system_other_odex u:object_r:exported3_default_prop:s0 exact int
+ro.crypto.allow_encrypt_override u:object_r:exported2_vold_prop:s0 exact bool
 ro.crypto.scrypt_params u:object_r:exported2_vold_prop:s0 exact string
+ro.crypto.volume.filenames_mode u:object_r:exported2_vold_prop:s0 exact string
 ro.dalvik.vm.native.bridge u:object_r:exported_dalvik_prop:s0 exact string
 ro.enable_boot_charger_mode u:object_r:exported3_default_prop:s0 exact bool
 ro.gfx.driver.0 u:object_r:exported3_default_prop:s0 exact string
diff --git a/public/vold.te b/public/vold.te
index 636e48fb..57171981 100644
--- a/public/vold.te
+++ b/public/vold.te
@@ -208,6 +208,9 @@ allow vold user_profile_data_file:dir create_dir_perms;
 # Raw writes to misc block device
 allow vold misc_block_device:blk_file w_file_perms;
 
+# vold might need to search or mount /mnt/vendor/*
+allow vold mnt_vendor_file:dir search;
+
 neverallow {
     domain
     -vold
-- 
2.17.1

